src/server.c

```diff
@@ -5678,6 +5682,7 @@ sds genValkeyInfoString(dict *section_dict, int all_sections, int everything) {
             "io_threaded_writes_processed:%lld\r\n", server.stat_io_writes_processed,
             "io_threaded_freed_objects:%lld\r\n", server.stat_io_freed_objects,
             "io_threaded_poll_processed:%lld\r\n", server.stat_poll_processed_by_io_threads,
+            "io_threaded_average_prefetch_batch_size:%lld\r\n", average_prefetch_batch_size,
```

Comment:
Generally averages are bad for info metrics because they don't give metrics that are useful "now". If the average batch size grew linearly from 1->10 for example, the current value would be 5, which is not that useful since it's not currently useful. My preference would be to expose both prefetch entries and batches so end users can compute an average on their own time frame OR change this to be a rolling average like the other instantaneous metrics.

Also, what do we see as the utility of this metric. You can't really turn this feature off, so would an end user have any idea what to do with this information?

Reply:
It's a good point that disabling it also makes these more useful, and maybe more actionable as well to tune the prefetch value.